aptos-move/framework/aptos-framework/sources/staking_contract.move

```diff
@@ -215,13 +215,49 @@ module aptos_framework::staking_contract {
     #[view]
     /// Return the number of pending distributions (e.g. commission, withdrawals from stakers).
     ///
-    /// This errors out the staking contract with the provided staker and operator doesn't exist.
+    /// This errors out if the staking contract with the provided staker and operator doesn't exist.
     public fun pending_distribution_counts(staker: address, operator: address): u64 acquires Store {
         assert_staking_contract_exists(staker, operator);
         let staking_contracts = &borrow_global<Store>(staker).staking_contracts;
         pool_u64::shareholders_count(&simple_map::borrow(staking_contracts, &operator).distribution_pool)
     }
 
+    #[view]
+    /// Return the amount that will be split between all stakers the next time distribute() is called.
+    ///
+    /// This errors out if the staking contract with the provided staker and operator doesn't exist.
+    public fun total_distributable_amount(staker: address, operator: address): u64 acquires Store {
+        assert_staking_contract_exists(staker, operator);
+        let store = borrow_global_mut<Store>(staker);
+        let staking_contract = simple_map::borrow(&store.staking_contracts, &operator);
+        let pool_address = staking_contract.pool_address;
+        let (_, inactive, _, pending_inactive) = stake::get_stake(pool_address);
+        inactive + pending_inactive
+    }
+
+    #[view]
+    /// Return the amount that will be disbursed to the staker the next time distribute() is called.
+    ///
+    /// This errors out if the staking contract with the provided staker and operator doesn't exist.
+    public fun distributable_amount(staker: address, operator: address): u64 acquires Store {
+        let total_distributable_amount = total_distributable_amount(staker, copy operator);
+        assert_staking_contract_exists(staker, operator);
+        let store = borrow_global_mut<Store>(staker);
+        let staking_contract = simple_map::borrow(&store.staking_contracts, &operator);
+        let shares = pool_u64::shares(&staking_contract.distribution_pool, staker);     
+        let new_staker_balance =
+            pool_u64::shares_to_amount_with_total_coins(
+                &staking_contract.distribution_pool, 
+                shares, 
+                total_distributable_amount
+            );
+
+        let old_staker_balance = pool_u64::balance(&staking_contract.distribution_pool, staker);
+        let commission = 
+            (new_staker_balance - old_staker_balance) * staking_contract.commission_percentage / 100;
+        new_staker_balance - commission
```

Comment:
We need to fully follow the logic in `update_distribution_pool`. Here we call `pool_u64::amount_to_shares_with_total_coins` which may bring precision error. So the equivalent amount might have a slight difference from `commission`.